File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/documents/inmemory/DocumentInfoRepositoryInMemory.java
package io.zenwave360.example.clinicaltool.modules.documents.inmemory;

import static org.apache.commons.lang3.ObjectUtils.firstNonNull;

import io.zenwave360.example.clinicaltool.modules.documents.DocumentInfoRepository;
import io.zenwave360.example.clinicaltool.modules.documents.domain.*;
import java.math.*;
import java.time.*;
import java.util.*;

public class DocumentInfoRepositoryInMemory extends InMemoryJpaRepository<DocumentInfo>
        implements DocumentInfoRepository {
    @Override
    public <T extends DocumentInfo> T save(T entity) {
        var finalEntity = super.save(entity);
        finalEntity
                .getDocumentData()
                .setId(firstNonNull(finalEntity.getDocumentData().getId(), finalEntity.getId()));
        return (T) finalEntity;
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/documents/inmemory/DocumentInfoRepositoryInMemory.kt
package io.zenwave360.example.clinicaltool.modules.documents.inmemory

import io.zenwave360.example.clinicaltool.modules.documents.DocumentInfoRepository
import io.zenwave360.example.clinicaltool.modules.documents.domain.*
import java.math.*
import java.time.*
import java.util.*

class DocumentInfoRepositoryInMemory : InMemoryJpaRepository<DocumentInfo, Long>(), DocumentInfoRepository {
    override fun <S : DocumentInfo> save(entity: S): S {
        super.save(entity)
        entity.documentData?.id = entity.documentData?.id ?: entity.id
        return entity
    }
}
